File: esp3d-starlight/scripts/fix_admin_pwd_mdx.py
from pathlib import Path

root = Path(__file__).resolve().parent.parent / 'src' / 'content' / 'docs' / 'ESP3D' / 'version-3x'
if not root.exists():
    raise SystemExit(f'Missing root: {root}')

pattern = '* pwd=<admin password>'
replacement = '* `pwd=<admin password>`'
count = 0
for path in root.rglob('*.mdx'):
    text = path.read_text(encoding='utf-8')
    if pattern in text:
        text = text.replace(pattern, replacement)
        path.write_text(text, encoding='utf-8')
        count += 1
print(f'Updated {count} files.')

File: esp3d-starlight/scripts/fix_admin_pwd_mdx.js
const fs = require('fs');
const path = require('path');
const root = path.resolve(__dirname, '..', 'src', 'content', 'docs', 'ESP3D', 'version-3x');
const pattern = '* pwd=<admin password>';
const replacement = '* `pwd=<admin password>`';
let count = 0;
function walk(dir) {
  for (const name of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, name.name);
    if (name.isDirectory()) walk(full);
    else if (name.isFile() && full.endsWith('.mdx')) {
      let text = fs.readFileSync(full, 'utf8');
      if (text.includes(pattern)) {
        text = text.split(pattern).join(replacement);
        fs.writeFileSync(full, text, 'utf8');
        count += 1;
      }
    }
  }
}
walk(root);
console.log('Updated', count, 'files.');
